test/tools/ossfuzz/proto_to_yul.cpp

```diff
@@ -0,0 +1,234 @@
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/tools/ossfuzz/proto_to_yul.h>
+#include <test/tools/ossfuzz/yul_proto.pb.h>
+
+#include <ostream>
+#include <sstream>
+
+namespace yul_fuzzer
+{
+// Forward decls.
+std::ostream& operator<<(std::ostream& _os, const BinaryOp& _x);
+
+std::ostream& operator<<(std::ostream& _os, const StatementSeq& _x);
+
+// Proto to yul.
+// Constant is a signed 32-bit integer
+std::ostream& operator<<(std::ostream& _os, const Const& _x)
+{
+	return _os << "(" << _x.val() << ")";
+}
+
+std::ostream& operator<<(std::ostream& _os , const VarRef& _x)
+{
+	return _os  << "x_" << (static_cast<uint32_t>(_x.varnum()) % 10);
+}
+
+std::ostream& operator<<(std::ostream& _os , const Lvalue& _x)
+{
+	return _os  << _x.varref();
+}
+
+std::ostream& operator<<(std::ostream& _os , const Rvalue& _x)
+{
+	if (_x.has_varref())
+		return _os  << _x.varref();
+	if (_x.has_cons())
+		return _os  << _x.cons();
+	if (_x.has_binop())
+		return _os  << _x.binop();
+	return _os  << "1";
+}
+
+std::ostream& operator<<(std::ostream& _os , const BinaryOp& _x)
+{
+	switch (_x.op()) {
+		case BinaryOp::ADD:
+			_os << "add(" << _x.left() << "," << _x.right() << ")";
+			break;
+		case BinaryOp::SUB:
+			_os << "sub(" << _x.left() << "," << _x.right() << ")";
+			break;
+		case BinaryOp::MUL:
+			_os << "mul(" << _x.left() << "," << _x.right() << ")";
+			break;
+		case BinaryOp::DIV:
+			_os << "div(" << _x.left() << "," << _x.right() << ")";
+			break;
+		case BinaryOp::MOD:
+			_os << "mod(" << _x.left() << "," << _x.right() << ")";
+			break;
+		case BinaryOp::XOR:
+			_os << "xor(" << _x.left() << "," << _x.right() << ")";
+			break;
+		case BinaryOp::AND:
+			_os << "and(" << _x.left() << "," << _x.right() << ")";
+			break;
+		case BinaryOp::OR:
+			_os << "or(" << _x.left() << "," << _x.right() << ")";
+			break;
+		case BinaryOp::EQ:
+			_os << "eq(" << _x.left() << "," << _x.right() << ")";
+			break;
+		case BinaryOp::LT:
+			_os << "lt(" << _x.left() << "," << _x.right() << ")";
+			break;
+		case BinaryOp::GT:
+			_os << "gt(" << _x.left() << "," << _x.right() << ")";
+			break;
+	}
+	return _os;
+}
+
+std::ostream& operator<<(std::ostream& _os , const VarDecl& _x)
+{
+	_os << "let x_" << _x.id() << " := " << _x.expr() << "\n";
+	return _os;
+}
+
+std::ostream& operator<<(std::ostream& _os , const TypedVarDecl& _x)
+{
+	_os << "let x_" << _x.id();
+	switch (_x.type())
+	{
+		case TypedVarDecl::BOOL:
+			_os << ": bool := " << _x.expr() << " : bool\n";
```

Comment:
If we had a function that would turn the enum into a string, this could also be greatly simplified.

Reply:
Won't a separate function for this be overkill? Atm, we only need it for Typed var decl statements.

Reply:
I thought protobuf might provide that.

Reply:
sadly not